harmony/harmony.py

```diff
@@ -361,6 +362,49 @@ def _params(self, request: Request) -> dict:
 
         return params
 
+    def _headers(self) -> dict:
+        """
+        Create (if needed) and return a dictionary of headers.
+        Code partially adaped from:
+            https://github.com/requests/toolbelt/blob/master/requests_toolbelt/utils/user_agent.py
+        """
+        if 'headers' not in self.__dict__:
+            user_agent_content = set([])
+
+            # Get harmony package info
+            try:
+                import harmony
+                user_agent_content.add(f'harmony-py/{harmony.__version__}')
+            except Exception as e:
+                print("Following exception was caught "
+                      "when building user-agent headers for harmony-py:")
+                print(e)
+
+            # Get platform info
+            try:
+                p_system = platform.system()
+                p_release = platform.release()
+                user_agent_content.add(f'{p_system}/{p_release}')
+            except Exception as e:
+                print("Following exception was caught "
+                      "when building user-agent headers for harmony-py:")
+                print(e)
+
+            # Get implementation info
+            try:
+                implementation = platform.python_implementation()
+                implementation_version = platform.python_version()
+                user_agent_content.add(f'{implementation}/{implementation_version}')
+            except Exception as e:
+                print("Following exception was caught "
+                      "when building user-agent headers for harmony-py:")
```

Comment:
I'm not sure it is a good idea to swallow the exceptions like this. In any case you probably don't want to return a possibly incorrect header on line 404.

Reply:
Well, if an exception is caught, it will not be saved to the user agent header (`user_agent_content`).
But a good point is that, if all header items failed, it probably doesn't make sense to have a empty header in line 404.

Reply:
Ok, a new commit pushed to handle empty user-agent headers; meanwhile, I found I didn't merge the existing user-agent headers, so provide the fix in the same commit.